simulator/modules/branch/branch.h

```diff
@@ -45,6 +45,22 @@ class Branch : public Log
         auto get_mispredict_rate() const { return ( num_branches != 0) 
                                                   ? num_mispredictions / num_branches * 100
                                                   : 0; }
+        template<typename Instr>
```

Comment:
If you have this function inside class, you may omit templatization, as type is known already